resteasy-jaxrs/src/main/java/org/jboss/resteasy/plugins/delegates/ClientCookieHeaderDelegate.java

```diff
@@ -0,0 +1,29 @@
+package org.jboss.resteasy.plugins.delegates;
+
+import javax.ws.rs.core.NewCookie;
+
+import org.jboss.resteasy.resteasy_jaxrs.i18n.Messages;
+
+/**
+ * 
+ * @author <a href="mailto:ron.sigal@jboss.com">Ron Sigal</a>
+ * @date May 20, 2016
+ */
+public class ClientCookieHeaderDelegate extends NewCookieHeaderDelegate
+{
+   public String toString(Object value)
+   {
+      if (value == null)
+      {
+         throw new IllegalArgumentException(Messages.MESSAGES.paramNull());
+      }
+      NewCookie cookie = (NewCookie) value;
+      StringBuilder b = new StringBuilder();
+      b.append(cookie.getName()).append('=');
+      if (cookie.getValue() != null)
+      {
+         quote(b, cookie.getValue());
```

Comment:
Hmmm. Well, that explains some of my confusion. I didn't know about this wrinkle. Actually, the javax.ws.rs.core.Cookie javadoc refers to RFC 2109.